File: moonpoison/linked_list/2130.cpp
#include<bits/stdc++.h>
using namespace std;
struct ListNode 
{
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};
class Solution {
public:
    int pairSum(ListNode* head) 
    {
        int mx=0;
        vector<int> v;
        while(head)
        {
            v.push_back(head->val);
            head=head->next;
        }
        for(int i=0;i<v.size();++i)
            mx=max(mx, v[v.size()-1-i]+v[i]);
        return mx;
    }
};

File: ice0208/linked-list/2130.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def pairSum(self, head):
        length = 1

        cur = head
        while cur.next:
            length += 1
            cur = cur.next

        twin_sum = []

        cur = head
        i = 0
        while cur:
            if i < length // 2:
                twin_sum.append(cur.val)
            else:
                twin_sum[length - i - 1] += cur.val

            cur = cur.next
            i += 1

        answer = max(twin_sum)

        return answer
